src/Workspaces/Core/Portable/SymbolId/SymbolKey.SymbolKeyComparer.cs

```diff
@@ -18,12 +18,12 @@ private SymbolKeyComparer(ComparisonOptions options)
 
             public bool Equals(SymbolKey x, SymbolKey y)
             {
-                if ((object)x == (object)y)
+                if (x == (object)y)
```

Comment:
nope.  The cast is redundant.  only one side needs to be object.  honestly though, i'd rather the code call ReferenceEquals